src/components/Container.js

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { Redirect, Route, Switch, useLocation } from 'react-router-dom';
+import { Locate, Details, Home, Search } from '../pages';
+import { useSearchQuery, useUserLocation } from '../hooks';
+
+
+const Container = () => {
+  const [query, setQuery] = useState('');
+  const { pathname } = useLocation();
+  const [skip, setSkip] = useState(pathname !== '/locate');
+  const locationForecast = useUserLocation(skip);
+  const searchForecast = useSearchQuery(query);
+  const id = pathname.split('/')[2];
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/megwali/punny-weather/pullRequest?prid=7693782)
```suggestion
  const id = pathname.split("/")[2];
```